mbed_targets/mbed_targets.py

```diff
@@ -90,6 +91,26 @@ def from_offline_target_entry(cls, target_entry: dict) -> "MbedTarget":
             build_variant=tuple(target_entry.get("build_variant", [])),
         )
 
+    def fetch_target_build_attributes(self, path_to_targets_json: str) -> Any:
+        """When provided with path to a targets.json file in Mbed OS, returns a dict of attributes.
+
+        These attributes contains the specific information needed to build Mbed applications for this Mbed Target.
+
+        Args:
+            path_to_targets_json: path to a targets.json file found in the Mbed OS library
+
+        Returns:
+            A dict containing the parsed attributes from targets.json
+
+        Raises:
+            FileNotFoundError: path provided does not lead to targets.json
```

Comment:
It looks like we suppress the `FileNotFoundError` and raise `TargetBuildAttributesError`, so `FileNotFoundError` isn't really relevant to the API here.

Reply:
56383a8